agnostic-svelte-ts/src/lib/components/Tabs/api.ts

```diff
@@ -0,0 +1,13 @@
+import type { SvelteComponentTyped } from 'svelte';
```

Comment:
Noice, I wasn't aware of `SvelteComponentTyped` 👍

Reply:
I was thinking a lot about how to type the components that can be passed to tabs. For now I've used `SvelteComponentTyped` because it seems like a good base class to create the external api of the component (or in this case, the component parameter).

They give a very apt example to use it:

> Base class to create strongly typed Svelte components.
> 
> This only exists for typing purposes and should be used in `.d.ts` files.
> 
>  ### Example:
> 
>  You have component library on npm called `component-library`, from which
>  you export a component called `MyComponent`. For Svelte+TypeScript users,
>  you want to provide typings. Therefore you create a `index.d.ts`:
>  ```ts
>  import { SvelteComponentTyped } from "svelte";
>  export class MyComponent extends SvelteComponentTyped<{foo: string}> {}
>  ```
>  Typing this makes it possible for IDEs like VS Code with the Svelte extension
>  to provide intellisense and to use the component like this in a Svelte file
>  with TypeScript:
>  ```svelte
>  <script lang="ts">
>  	import { MyComponent } from "component-library";
>  </script>
>  <MyComponent foo={'bar'} />
>  ```